doc/nrf/libraries/nrf_security/doc/mbed_tls_header.rst

```diff
@@ -3,23 +3,30 @@
 User-provided Mbed TLS config header
 ####################################
 
-The Nordic Security Module provides a Kconfig interface to control compilation and linking of Mbed TLS and the :ref:`nrf_cc3xx_mbedcrypto_readme` or :ref:`nrf_oberon_readme` libraries.
+The Nordic Security Module provides a Kconfig and CMake based build system to configure the Mbed TLS, :ref:`nrf_cc3xx_mbedcrypto_readme`, and :ref:`nrf_oberon_readme` libraries.
 
-The Kconfig interface and build system ensures that the configuration of nrf_security is valid and working.
-It also ensures that dependencies between different cryptographic APIs are met.
+It is recommended to let this system generate the Mbed TLS configuration header as it enforces dependencies between the libraries.
 
-It is therefore highly recommended to let the build system generate the Mbed TLS configuration headers.
+However, for special use cases that cannot be configured with this system it is possible to provide custom Mbed TLS configuration headers.
 
-However, for special use cases that cannot be achieved using the Kconfig configuration tool, it is possible to provide custom Mbed TLS configuration headers.
+This can be done with the below procedure:
 
-Make sure that the system is working:
-
-1. Use Kconfig and the build system to create Mbed TLS configuration headers as a starting point.
-#. Edit this file to include settings that are not available in Kconfig.
+1. Generate the Mbed TLS configuration header
+#. Copy the Mbed TLS header named nrf-config.h out of the build directory and give it a custom name
+#. In the project configuration make the following changes:
+#. CONFIG_GENERATE_MBEDTLS_CFG_FILE=n
+#. CONFIG_MBEDTLS_CFG_FILE="custom-name-nrf-config.h"
+#. CONFIG_MBEDTLS_USER_CONFIG_FILE="empty_file.h"
+#. Create an empty file named empty_file.h
+#. Edit custom-name-nrf-config.h with custom configuration
+#. If not already on path, add the header files to the include path by editing the application build scripts
```

Comment:
```suggestion
1. Generate the Mbed TLS configuration header.
#. Create a copy of the :file:`nrf-config.h` Mbed TLS header and give it a custom name.
#. In the project configuration, make the following changes:

   * `CONFIG_GENERATE_MBEDTLS_CFG_FILE=n`
   * `CONFIG_MBEDTLS_CFG_FILE="custom-name-nrf-config.h"` where the file name 
   * `CONFIG_MBEDTLS_USER_CONFIG_FILE="empty_file.h"`
#. Create an empty file named :file:`empty_file.h`.
#. Edit :file:`custom-name-nrf-config.h` with your custom configuration.
#. If not already on path, add the header files to the include path by editing the application build scripts.
```

'Copy' would imply that the original stays in the same place. Step 2 should have some formatting around the file name, but I'm not sure what.

I don't really understand the last line.

Reply:
"out of the build directory" is omitted in this rewrite. I want to make it clear to the user where the file he needs to copy/move from the build dir is located.

Reply:
* `CONFIG_MBEDTLS_CFG_FILE="custom-name-nrf-config.h"` where the file name 

I assume "where the file name" should be omitted?

Reply:
"I don't really understand the last line."

How about:

If the header files are not already on the include path, then add them to path by editing the application build scripts

Reply:
> * `CONFIG_MBEDTLS_CFG_FILE="custom-name-nrf-config.h"` where the file name
> 
> I assume "where the file name" should be omitted?

Ah. Sorry, yes.

Reply:
> If the header files are not already on the include path, add them to path by editing the application build scripts.

Reply:
"Is the file they need to copy in the build directory, or do they need to save a copy outside of the build directory?"

They need to either move or copy (and rename) a file that is in the build directory into the source directory.

Reply:
mv build/file.c src/my-file.c
cp build/file.c src/my-file.c

either of these operations work

Reply:
So:

#. Create a copy of the :file:`nrf-config.h` Mbed TLS header file in the build directory and give it a custom name.
#. Move this custom Mbed TLS header file to your source directory.

Something like that?

Reply:
sure

Reply:
applied fix